### PR TITLE
MB-12099 Refactor parsing for History API

### PR DESCRIPTION
**Pull request by YanZ777**
## Summary

Most of the work was already done by [MB-12098](https://dp3.atlassian.net/browse/MB-12098) already.

PropTypes.object isn't supposed to be used so this swaps out the PropTypes.object with the shape of what the object would look like.
The linter typically doesn't like PropTypes.object. I think the only reason it wasn't flagged here was because it wasn't directly in the prop types for a component.

A point was also brought up [here](https://github.com/transcom/mymove/pull/8368#discussion_r841958018) where maybe we are ok with just having PropTypes.object though? 

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make office_client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the office app
2. Login as a TOO
3. Make sure everything still works.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

### CODE REVIEW

File: src/constants/historyLogUIDisplayName.js [hunk 1]
```diff
@@ -1,6 +1,25 @@
 import PropTypes from 'prop-types';
 
-export const HistoryLogValuesShape = PropTypes.object;
+export const HistoryLogValuesShape = PropTypes.shape({
+  updated_dat: PropTypes.string,
```
Comment (donaldthai) [inline src/constants/historyLogUIDisplayName.js:4]: nit: `updated_dat` should be `updated_at`

File: src/constants/historyLogUIDisplayName.js [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+  diversion: PropTypes.string,
+  billable_weight_cap: PropTypes.string,
+  uses_external_vendor: PropTypes.string,
+  requested_delivery_date: PropTypes.string,
+  scheduled_pickup_date: PropTypes.string,
+  status: PropTypes.string,
+  customer_remarks: PropTypes.string,
+  approved_dat: PropTypes.string,
```
Comment (donaldthai) [inline src/constants/historyLogUIDisplayName.js:12]: nit: `approved_dat` should be `approved_at`

**Review by donaldthai — APPROVED** [review]
Looks good. Made some comments on some naming.

**Comment by YanZ777** [comment]
@kennethchow-truss I saw [your comment here](https://github.com/transcom/mymove/pull/8368#discussion_r841958018), so I'm not sure if we should actually merge this PR?

**Comment by YanZ777** [comment]
Closing this PR after discussing with Kenneth and agreeing that there's too many fields to list for this to be useful.

**Pull request closed without merging**